src/persistence.rs

```diff
@@ -246,26 +235,26 @@ pub mod embedding {
     impl EmbeddingPersistor for TextFileVectorPersistor {
         fn put_metadata(&mut self, entity_count: u32, dimension: u16) {
             let metadata = format!("{} {}", entity_count, dimension);
-            self.buf_writer.write(metadata.as_bytes());
+            self.buf_writer.write(metadata.as_bytes()).ok();
```

Comment:
I wonder if we can do something else here and in the `finish`. `ok` just covers up the problem.

Reply:
Possibly, but this is unrelated to my PR.